### PR TITLE
feat(smart-contracts): add proposal to redeploy gov

### LINKED ISSUE #12015: Redeploy Governor contract (upgradable)
- [x] Redeploy the actual Governor contract for UDT on mainnet with upgraded OZ deps, using the existing timelock
- [x] Send 1 proposal to the current Governor that contains
   - [x] add the new Governor as proposer in the existing timelock 
   - [x] renounce proposer role for the current Governor

### PR DESCRIPTION
**Pull request by clemsos**
# Description

Script for the proposal to redeploy gov - fix #12015

NB: this requires https://github.com/unlock-protocol/unlock/pull/12065 to work

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
We should probably nerge https://github.com/unlock-protocol/unlock/pull/11673 before?

File: smart-contracts/proposals/005-redeploy-governor.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    contractAddress: timeLockAddress,
+    functionName: 'revokeRole',
+    functionArgs: [TIMELOCK_ADMIN_ROLE, oldGovAddress],
+  }
+
+  const proposalArgs = {
+    calls: [proposalGrantRoleArgs, proposalRevokeRoleArgs],
+    proposalName: 'Switch Timelock admin role to the new Governor',
+  }
+  console.log(proposalArgs)
```
Comment (julien51) [inline smart-contracts/proposals/005-redeploy-governor.js:36]:
```suggestion
```

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
@clemsos there was a security upgrade by OZ so I rebased this one!

File: smart-contracts/proposals/005-redeploy-governor.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      contractAddress: timeLockAddress,
+      functionName: 'revokeRole',
+      functionArgs: [TIMELOCK_ADMIN_ROLE, oldGovAddress],
+    },
+    {
+      contractName: 'UnlockProtocolTimelock',
+      contractAddress: timeLockAddress,
+      functionName: 'revokeRole',
+      functionArgs: [PROPOSER_ROLE, oldGovAddress],
+    },
```
Comment (clemsos) [inline smart-contracts/proposals/005-redeploy-governor.js:86]: We could also add a "canceller" role here.

Comment (julien51) [reply smart-contracts/proposals/005-redeploy-governor.js:86]: ha! ok

**Review by julien51 — APPROVED** [review]

**Comment by clemsos** [comment]
should rebase when #12256 is merged and then we are good to merge as the proposal has been submitted.

**Pull request merged**